### PR TITLE
Implement jsonb containment operators

### PR DESCRIPTION
**Pull request by zaneduffield**
Implemented jsonb containment operators `<@` and `@>` as described in #3032.
First time contributing so please forgive any mistakes.

### CODE REVIEW

File: diesel/src/pg/expression/expression_methods.rs [hunk 1]
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
...
+    /// ```
+    fn contains<T>(self, other: T) -> dsl::ContainsJsonb<Self, T>
+    where
+        T: AsExpression<Jsonb>,
+    {
+        Grouped(ContainsJsonb::new(self, other.as_expression()))
+    }
+    /// Creates a PostgreSQL `<@` expression.
+    ///
+    /// This operator returns whether a JSON value is contained by another JSON value. `foo.contains(bar)` is the
```
Comment (pksunkara) [inline diesel/src/pg/expression/expression_methods.rs:1365]:
```suggestion
    /// This operator checks whether left hand side JSONB value is contained by right hand side JSON value. `foo.contains(bar)` is the
```

Please break up the sentence into lines accordingly after the length changes.

**Review by pksunkara — CHANGES_REQUESTED** [review]

File: diesel/src/pg/expression/expression_methods.rs [hunk 2]
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
     {
         Grouped(HasKeyJsonb::new(self, other.as_expression()))
     }
+    /// Creates a PostgreSQL `@>` expression.
+    ///
+    /// This operator returns whether a JSON value contains another JSON value
```
[the remark this answers is not in the document]
Comment (zaneduffield) [reply diesel/src/pg/expression/expression_methods.rs]: fixed

File: diesel/src/pg/expression/expression_methods.rs [hunk 3]
```diff
@@ -1301,6 +1301,129 @@ pub trait PgJsonbExpressionMethods: Expression + Sized {
...
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn contains<T>(self, other: T) -> dsl::ContainsJsonb<Self, T>
+    where
+        T: AsExpression<Jsonb>,
+    {
+        Grouped(ContainsJsonb::new(self, other.as_expression()))
+    }
+    /// Creates a PostgreSQL `<@` expression.
```
[the remark this answers is not in the document]
Comment (zaneduffield) [reply diesel/src/pg/expression/expression_methods.rs:1365]: fixed

File: diesel/src/pg/expression/expression_methods.rs (hunk 1 above)
Comment (zaneduffield) [reply diesel/src/pg/expression/expression_methods.rs]: fixed

**Comment by zaneduffield** [comment]
I'm not sure why that "sqlite bundled" test is failing now. The error makes it seem unlreated to my changes.

**Comment by weiznich** [comment]
Yes, that's an unrelated CI failure. Likely due to a updated nightly version that introduced a new warning.

**Review by pksunkara — CHANGES_REQUESTED** [review]
My comments are still not addressed. Are you sure you pushed correctly?

**Comment by zaneduffield** [comment]
> My comments are still not addressed. Are you sure you pushed correctly?

It looks like I somehow reverted my changes which addressed them in the merge commit. I'll fix this when I get home today.

**Comment by zaneduffield** [comment]
I fixed the issue, but also realised that fixup commits on merge commits require you to rebase and do the merge again. To clean up this mess a bit I've rebased to latest master, resolving merge conflicts, and force pushed to this branch.

**Review by pksunkara — APPROVED** [review]

**Comment by pksunkara** [comment]
@weiznich not sure if you want to merge or not because of that SQLite issue

**Comment by weiznich** [comment]
It's fine to merge it, I will try to address the sqlite issue today or tomorrow in a separate PR.

**Pull request merged**